src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxTests.cs

```diff
@@ -454,6 +468,65 @@ public static void CollectionPerphemeralImport_HasKeyName()
             }
         }
 
+        [ConditionalTheory]
+        [MemberData(memberName: nameof(PfxIterationCountTests.GetCertsWith_IterationCountNotExceedingDefaultLimit_AndNullOrEmptyPassword_MemberData), MemberType = typeof(PfxIterationCountTests))]
+        public static void TestIterationCounter(string name, bool usesPbes2, byte[] blob, int iterationCount)
+        {
+            _ = iterationCount;
+
+            MethodInfo method = typeof(X509Certificate).GetMethod("GetIterationCount", BindingFlags.Static | BindingFlags.NonPublic);
+            GetIterationCountDelegate target = method.CreateDelegate<GetIterationCountDelegate>();
+
+            if (usesPbes2 && !Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            try
+            {
+                long count = (long)target(blob, out int bytesConsumed);
+                Assert.Equal(iterationCount, count);
+                Assert.Equal(blob.Length, bytesConsumed); // we currently don't have any cert with trailing data.
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"There's an error on certificate {name}, see inner exception for details", e);
+            }
+        }
+
+        internal static bool IsPkcs12IterationCountAllowed(long iterationCount, long allowedIterations)
+        {
+            if (allowedIterations == UnlimitedIterations)
+            {
+                return true;
+            }
+
+            if (allowedIterations == UnspecifiedIterations)
+            {
+                allowedIterations = DefaultIterations;
+            }
+
+            Assert.True(allowedIterations >= 0);
+
+            return iterationCount <= allowedIterations;
+        }
+
+        // This is a horrible way to produce SecureString. SecureString is deprecated and should not be used.
+        // This is only reasonable because it is a test driver.
+        internal static SecureString? GetSecureString(string password)
```

Comment:
Since a null input is handled, it seems like the correctly annotated signature is

```suggestion
        internal static SecureString? GetSecureString(string? password)
```

Reply:
I thought all of the tests were null-oblivious. I would not have expected anything to be annotated.

Reply:
Zero question marks is also fine :)